cmd/kubeadm/app/phases/addons/dns/manifests.go

```diff
@@ -129,11 +133,11 @@ spec:
         securityContext:
           allowPrivilegeEscalation: false
           capabilities:
-            add:
-            - NET_BIND_SERVICE
             drop:
-            - all
+            - ALL
           readOnlyRootFilesystem: true
+          runAsGroup: 2000
+          runAsUser: 1000
```

Comment:
I don't think we should be running this with an arbitrary user / group until we reserve them the same way we did for the control plane - RootlessControlPlane.

There is also a binding problem here. If coredns in kubeadm moves to an operator the user / group reservation must be propageted to the coredns container that the operator manages.

Reply:
Just an idea. We may do some alpha support for this in 1.24.

RootlessControlPlane is still alpha now. And we may add this and reserve the user/group for coredns like control-plane and using this safe sysctl if RootlessControlPlane is enabled in 1.24.

In the future release when we want to graduate RootlessControlPlane to beta, we can get more feedback during that time.

Reply:
There is a kep in k/enhancements for Linux "user namespaces" which from my understanding renders our kubeadm manual user/group assignment redundant.

For now the plan was to graduate the rootless kubeadm mode to beta in 1.24. But i guess we need to discuss it more..

Reply:
AFAIK the user namespaces kep alpha only applies to containers that don't mount hostVolumes.

Reply:
[User namespace](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/127-user-namespaces#phase-2-pods-with-volumes
) phase 1 is ready in v1.25 for pod.
- Phase 2: pods with volumes.

Reply:
`RootlessControlPlane` is out of the topic of the initial PR.